x-pack/plugin/eql/src/test/resources/querytranslator_tests.txt

```diff
@@ -804,16 +828,99 @@ process where command_line regex~ ("^.*?net.exe", "net\\.exe", "C:\\\\Windows\\\
+;
+
+// the conjunction will be applied on different values of the multi-value field
+conjunctionOfSameFieldSameFunction
+process where length(file_name) > 5 and length(file_name) < 10
+;
+{"bool":{"must":[{"script":{"script":{"source":"InternalEqlScriptUtils.multiValueDocValues(doc,params.v0,X0->InternalQlScriptUtils.nullSafeFilter(InternalQlScriptUtils.gt(InternalEqlScriptUtils.length(X0),params.v1)))"
+"params":{"v0":"file_name.keyword","v1":5}}
+{"script":{"script":{"source":"InternalEqlScriptUtils.multiValueDocValues(doc,params.v0,X0->InternalQlScriptUtils.nullSafeFilter(InternalQlScriptUtils.lt(InternalEqlScriptUtils.length(X0),params.v1)))"
+"params":{"v0":"file_name.keyword","v1":10}}
+;
+
+// the disjunction will be applied on different values of the multi-value field
+disjunctionOfSameFieldSameFunction
+process where length(file_name) > 5 or length(file_name) < 10
+;
+{"bool":{"should":[{"script":{"script":{"source":"InternalEqlScriptUtils.multiValueDocValues(doc,params.v0,X0->InternalQlScriptUtils.nullSafeFilter(InternalQlScriptUtils.gt(InternalEqlScriptUtils.length(X0),params.v1)))"
+"params":{"v0":"file_name.keyword","v1":5}}
+{"script":{"script":{"source":"InternalEqlScriptUtils.multiValueDocValues(doc,params.v0,X0->InternalQlScriptUtils.nullSafeFilter(InternalQlScriptUtils.lt(InternalEqlScriptUtils.length(X0),params.v1)))"
+"params":{"v0":"file_name.keyword","v1":10}}
+;
+
+isNullInScript
+process where null == (exit_code > -1)
+;
+{"script":{"script":{"source":"InternalEqlScriptUtils.multiValueDocValues(doc,params.v0,X0->InternalQlScriptUtils.nullSafeFilter(InternalQlScriptUtils.isNull(InternalQlScriptUtils.gt(X0,params.v1))))"
+"params":{"v0":"exit_code","v1":-1}}
+;
+
+disjunctionOfFunctionAndNegatedFunction
+process where not (length(user) == 1) and length(user) == 1
+;
+{"bool":{"must":[{"script":{"script":{"source":"InternalQlScriptUtils.not(InternalEqlScriptUtils.multiValueDocValues(doc,params.v0,X0->InternalQlScriptUtils.nullSafeFilter(InternalQlScriptUtils.eq(InternalEqlScriptUtils.length(X0),params.v1))))"
+"params":{"v0":"user","v1":1}}
+{"script":{"script":{"source":"InternalEqlScriptUtils.multiValueDocValues(doc,params.v0,X0->InternalQlScriptUtils.nullSafeFilter(InternalQlScriptUtils.eq(InternalEqlScriptUtils.length(X0),params.v1)))"
+"params":{"v0":"user","v1":1}}
+;
+
+twoRangesOnASingleField
+// in a multi-value fields scenario where we want different values of the same field to obey the two conditions we need to use
+// two range queries, because a single range query will apply the two conditions on the same value of the multi-value field.
+process where pid > 100 and pid < 200
+;
+{"bool":{"must":[{"range":{"pid":{"from":100,"to":null,"include_lower":false,"include_upper":false
+{"range":{"pid":{"from":null,"to":200,"include_lower":false,"include_upper":false
+;
+
+// this type of query (disjunction/conjunction inside a function where one of the conditions is negated) is currently unsupported because
```

Comment:
works for me. in my other implementation, I treat any conjunctions inside a function as operating on a single value at a time. I think we have the same semantics. But I think it's also perfectly acceptable to solve that particular problem separately.